Editor/ExportUtility.cs

```diff
@@ -690,30 +709,62 @@ static void BakeNonstandardRenderers(DynamicObject rootDynamic, List<BakeableMes
-                bm.tempGo.transform.localPosition = Vector3.zero;
-                bm.meshRenderer = bm.tempGo.AddComponent<MeshRenderer>();
-                bm.meshRenderer.sharedMaterial = new Material(Shader.Find("Hidden/Cognitive/Canvas Export Shader")); //2 sided transparent diffuse
+                ExportQuad(v.gameObject, meshes, ExportQuadType.Canvas);
+            }
+        }
+
+        private static void ExportQuad(GameObject v, List<BakeableMesh> meshes, ExportQuadType type)
+        {
+            BakeableMesh bm = new BakeableMesh();
+            bm.tempGo = new GameObject(v.gameObject.name);
+            bm.tempGo.transform.parent = v.transform;
+            bm.tempGo.transform.localScale = Vector3.one;
+            bm.tempGo.transform.localRotation = Quaternion.identity;
+
+            //remove transform scale
+            float width = 0;
+            float height = 0;
+            bm.tempGo.transform.position = v.transform.position;
+            if (type == ExportQuadType.Canvas)
+            {
+                width = v.GetComponent<RectTransform>().sizeDelta.x;
+                height = v.GetComponent<RectTransform>().sizeDelta.y;
+                if (Mathf.Approximately(width, height))
+                {
+                    //centered
+                }
+                else if (height > width) //tall
+                {
+                    //half of the difference between width and height
+                    bm.tempGo.transform.position += (bm.tempGo.transform.right) * (height - width) / 2;
+                }
+                else //wide
+                {
+                    //half of the difference between width and height
+                    bm.tempGo.transform.position += (bm.tempGo.transform.up) * (width - height) / 2;
+                }
+            }
+            else if (type == ExportQuadType.TMPro)
+            {
+                MeshRenderer mr = v.GetComponent<MeshRenderer>();
+                width = mr.bounds.size.x;
+                height = mr.bounds.size.y;
+                bm.tempGo.transform.position = v.GetComponent<MeshRenderer>().bounds.center;
+            }
 
-                //remove transform scale
-                var width = v.GetComponent<RectTransform>().sizeDelta.x;// * v.transform.localScale.x;
-                var height = v.GetComponent<RectTransform>().sizeDelta.y;// * v.transform.localScale.y;
+            bm.meshRenderer = bm.tempGo.AddComponent<MeshRenderer>();
+            bm.meshRenderer.sharedMaterial = new Material(Shader.Find("Hidden/Cognitive/Canvas Export Shader")); //2 sided transparent diffuse
 
-                //bake texture from render
-                var screenshot = CanvasTextureBake(v.transform);
-                screenshot.name = v.gameObject.name.Replace(' ', '_');
-                bm.meshRenderer.sharedMaterial.mainTexture = screenshot;
+            //bake texture from render
+            var screenshot = TextureBake(v.transform, type, width, height);
```

Comment:
Function renamed from `CanvasTextureBake` to signify that it does more than just bake Canvases.